MetabolicAtlas/MetabolicAtlas
# feat: enzyme db details

**Related issue(s) and PR(s)**  
This PR closes #1014 .



**Type of change**  

 
**List of changes made**  

- Update PostgreSQL schema.
- Add API endpoints for enzymeDB, `enzyme` as well as supplementary data `compound`, `ec`, `reaction`.
- Add page in frontend that displays the entity (`compound`, `ec`, or `reaction`) details as well as listing the related `enzymes`.
- Update the cross references page to accommodate for IDs that are not found in the `neo4j` database.

**Screenshot of the fix**  

<img width="1579" alt="image" src="https://user-images.githubusercontent.com/423498/173382156-0c8a0a59-17e9-44bc-944a-65d5110c774f.png">
<img width="1522" alt="image" src="https://user-images.githubusercontent.com/423498/173381932-429e41fd-50ae-46d0-ad5b-789309794fdc.png">
<img width="637" alt="image" src="https://user-images.githubusercontent.com/423498/173382113-45bd2d65-3ddd-430c-9861-c8715a4c5c54.png">


**Testing**  

- A rebuild is needed due to new dependencies
- Relative urls that can be reused both for production and local testing
- Instructions on how to test

1. Visit an entity's details page. For example:
  - `/enzymedb/compound/C00002`
  - `/enzymedb/ec/2.5.1.18`
  - `/enzymedb/reaction/R00006`
2. Verify that the basic info and cross references (not available for `ec`) tables are displayed.
3. Verify that the enzymes table is displayed.
  - Please try out the filters (must be exact match), table sorting and pagination and verify that they work as expected.
4. Visit an entity's details page that does not exist (for example `/enzymedb/reaction/R000060`) and verify that and error pop-up is shown.

If you run into any problems with the database, please try resetting it. I also noticed that if you don't want any errors during the data import, it helps to delete the extra empty lines at the end of files. For me this was the `abac.txt` and `neq.txt` files.

To reset the database, please do a `git pull` and run the following commands.

```bash

ma-exec pg psql -f scripts/delete-db.sql -U postgres
ma-exec pg psql -f scripts/init-schema.sql -U postgres
ma-exec pg psql -f scripts/seed-data.sql -U postgres

```

**Further comments**  


**Definition of Done checklist**
